tests/TestCase/Command/SeedCommandTest.php

```diff
@@ -265,4 +265,120 @@ public function testSeederWithDateTimeFields(): void
         $this->assertNotEmpty($store['created']);
         $this->assertNotEmpty($store['modified']);
     }
+
+    public function testDryRunModeWarning(): void
+    {
+        $this->createTables();
+        $this->exec('migrations seed -c test --seed NumbersSeed --dry-run');
+
+        $this->assertExitSuccess();
+        $this->assertOutputContains('<warning>dry-run mode enabled</warning>');
+        $this->assertOutputContains('NumbersSeed:</info> <comment>seeding');
+        $this->assertOutputContains('All Done');
+    }
+
+    public function testDryRunModeShortOption(): void
+    {
+        $this->createTables();
+        $this->exec('migrations seed -c test --seed NumbersSeed -x');
+
+        $this->assertExitSuccess();
+        $this->assertOutputContains('<warning>dry-run mode enabled</warning>');
+        $this->assertOutputContains('NumbersSeed:</info> <comment>seeding');
+        $this->assertOutputContains('All Done');
+    }
+
+    public function testDryRunModeNoDataChanges(): void
+    {
+        $this->createTables();
+
+        /** @var \Cake\Database\Connection $connection */
+        $connection = ConnectionManager::get('test');
+        $initialCount = $connection->execute('SELECT COUNT(*) FROM numbers')->fetchColumn(0);
+
+        $this->exec('migrations seed -c test --seed NumbersSeed --dry-run');
+        $this->assertExitSuccess();
+
+        $finalCount = $connection->execute('SELECT COUNT(*) FROM numbers')->fetchColumn(0);
+        $this->assertEquals($initialCount, $finalCount, 'Dry-run mode should not modify database');
+    }
+
+    public function testDryRunModeMultipleSeeds(): void
+    {
+        $this->createTables();
+        $this->exec('migrations seed -c test --source CallSeeds --seed LettersSeed --seed NumbersCallSeed --dry-run');
+
+        $this->assertExitSuccess();
+        $this->assertOutputContains('<warning>dry-run mode enabled</warning>');
+        $this->assertOutputContains('NumbersCallSeed:</info> <comment>seeding');
+        $this->assertOutputContains('LettersSeed:</info> <comment>seeding');
+        $this->assertOutputContains('All Done');
+
+        /** @var \Cake\Database\Connection $connection */
+        $connection = ConnectionManager::get('test');
+        $numbersCount = $connection->execute('SELECT COUNT(*) FROM numbers')->fetchColumn(0);
+        $lettersCount = $connection->execute('SELECT COUNT(*) FROM letters')->fetchColumn(0);
+
+        $this->assertEquals(0, $numbersCount, 'Dry-run mode should not insert into numbers table');
+        $this->assertEquals(0, $lettersCount, 'Dry-run mode should not insert into letters table');
+    }
+
+    /**
+     * TODO: Currently failing with "TypeError: PDO::quote(): Argument #1 ($string) must be of type string, Cake\I18n\Date given"
```

Comment:
We might have an invalid seed in the test files. Could you use a `--source` to use a set of seeds that will work better?

Reply:
The seed is "valid" we added DateTime support
The handling is wrong somewhere as it then tried to work only with strings